prover/protocol/distributed/compiler/plonkinwizard/distribute.go

```diff
@@ -0,0 +1,69 @@
+package plonkinwizard
+
+import (
+	"github.com/consensys/linea-monorepo/prover/protocol/distributed"
+	"github.com/consensys/linea-monorepo/prover/protocol/query"
+	"github.com/consensys/linea-monorepo/prover/protocol/wizard"
+	"github.com/consensys/linea-monorepo/prover/utils"
+)
+
+// DistributePlonkInWizard scans initComp for non-ignored PlonkInWizard queries and
+// identifies the ones corresponding to moduleName. For each of those queries, the
+// function inserts a corresponding query into the module. The function will panic
+// if:
+//
+//   - the query is cross-module
+//
+// The new query is analog to the one in initComp but and shares the same name.
+func DistributePlonkInWizard(
+	moduleName distributed.ModuleName,
```

Comment:
Not directly about this PR, but seems that `distributed.ModuleName` is defined as an alias to `string`, this means that it is exacly the same as `string`. Wouldn't it be better to define it as a distinct type instead so we would have some type safety? A la `type ModuleName string` (note missing `=`)

Reply:
I agree, I am actually refactoring that in my current branch